uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java

```diff
@@ -11,96 +11,192 @@
+	@Parameters(name = "{0}")
+	public static Collection<FuseProjectDefinition> setupData() {
+		String camelVersion = serverRequirement.getConfiguration().getCamelVersion();
+		NewFuseIntegrationProjectWizardDeploymentType deploymentType = STANDALONE;
+		NewFuseIntegrationProjectWizardRuntimeType runtimeType;
+		switch (RUNTIME_TYPE) {
+		case "Karaf":
+			runtimeType = KARAF;
+			break;
+		case "EAP":
+			runtimeType = EAP;
+			break;
+		default:
+			runtimeType = KARAF;
+		}
+		NewFuseIntegrationProjectWizard wiz = new NewFuseIntegrationProjectWizard();
+		wiz.open();
+		NewFuseIntegrationProjectWizardFirstPage firstPage = new NewFuseIntegrationProjectWizardFirstPage(wiz);
+		firstPage.setProjectName(PROJECT_NAME);
+		wiz.next();
+		NewFuseIntegrationProjectWizardRuntimePage secondPage = new NewFuseIntegrationProjectWizardRuntimePage(wiz);
+		secondPage.setDeploymentType(deploymentType);
+		secondPage.setRuntimeType(runtimeType);
+		secondPage.typeCamelVersion(camelVersion);
+		wiz.next();
+		NewFuseIntegrationProjectWizardAdvancedPage lastPage = new NewFuseIntegrationProjectWizardAdvancedPage(wiz);
+		List<FuseProjectDefinition> projects = new ArrayList<>();
+		for (String[] template : lastPage.getAllAvailableTemplates()) {
+			ProjectType dsl;
+			if (template[template.length - 1].toLowerCase().contains("blueprint")) {
+				dsl = ProjectType.BLUEPRINT;
+			} else if (template[template.length - 1].toLowerCase().contains("spring")) {
+				dsl = ProjectType.SPRING;
+			} else {
+				dsl = ProjectType.JAVA;
+			}
+			FuseProjectDefinition tmp = new FuseProjectDefinition(runtimeType, deploymentType, template, camelVersion,
+					dsl);
+			projects.add(tmp);
 		}
+		wiz.cancel();
+		return projects;
 	}
 
 	/**
-	 * Cleans up test environment
+	 * Utilizes passing parameters using the constructor
+	 * 
+	 * @param template
+	 *                     a Fuse project archetype
 	 */
+	public DeploymentTest(FuseProjectDefinition project) {
+		this.project = project;
+	}
+
+	@Before
+	public void prepareWorkspace() {
+		new CleanErrorLogRequirement().fulfill();
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(project.getDeploymentType())
+				.runtimeType(project.getRuntimeType()).version(project.getCamelVersion())
+				.template(project.getTemplate()).create();
+	}
+
 	@After
-	public void setupManageServers() {
+	public void cleanWorkspace() {
+		new CleanWorkspaceRequirement().fulfill();
+	}
 
-		new WorkbenchShell().setFocus();
-		FuseServerManipulator.removeAllModules(serverRequirement.getConfiguration().getServer().getName());
-		FuseServerManipulator.stopServer(serverRequirement.getConfiguration().getServer().getName());
-		FuseServerManipulator.removeServer(serverRequirement.getConfiguration().getServer().getName());
+	@AfterClass
+	public static void cleanServers() {
+		String serverName = serverRequirement.getConfiguration().getServer().getName();
+		FuseServerManipulator.removeAllModules(serverName);
```

Comment:
we should use _FuseServerManipulator.removeAllModules(serverName);_ method also in _@After_ method. If something went wrong with the test, it ensures that the project will be undeployed and do not affect next test.